### PR TITLE
Support default action in P4TC

### PR DESCRIPTION
**Pull request by komaljai**
Support default table action
Fix hdrMd variable in controlBlock.c file
Move Parser declaration in .h file
Reintroduce adjust headroom

### CODE REVIEW

File: backends/tc/ebpfCodeGen.cpp [hunk 1]
```diff
@@ -893,6 +905,35 @@ void EBPFTablePNA::emitDefaultActionStruct(EBPF::CodeBuilder *builder) {
...
+void EBPFTablePNA::emitDefaultAction(EBPF::CodeBuilder *builder, cstring valueName) {
+    const IR::P4Table *t = table->container;
+    const IR::Expression *default_action = t->getDefaultAction();
+    bool visitDefaultAction = false;
+    if (default_action) {
+        if (auto mc = default_action->to<IR::MethodCallExpression>()) {
+            default_action = mc->method;
+        }
+        auto path = default_action->to<IR::PathExpression>();
+        BUG_CHECK(path, "Default action path %s cannot be found", default_action);
```
Comment (usha1830) [inline backends/tc/ebpfCodeGen.cpp:917]: %s ->%1%

File: testdata/p4tc_samples_outputs/default_action_example_control_blocks.c [hunk 2]
```diff
@@ -151,8 +147,8 @@ if (/* hdr->ipv4.isValid() */
                                 return TC_ACT_SHOT;
                         }
                     } else {
-                        return TC_ACT_SHOT;
-;
+/* drop_packet() */
```
Comment (usha1830) [inline testdata/p4tc_samples_outputs/default_action_example_control_blocks.c:150]: Indentation looks off in the c output

File: testdata/p4tc_samples_outputs/default_action_example_control_blocks.c [hunk 3]
```diff
@@ -224,7 +220,16 @@ if (/* hdr->ipv4.isValid() */
 ;        if (hdr->ipv4.ebpf_valid) {
             outHeaderLength += 160;
         }
-;        pkt = ((void*)(long)skb->data);
+;
```
Comment (usha1830) [inline testdata/p4tc_samples_outputs/default_action_example_control_blocks.c:223]: empty statement. can be removed?

**Review by usha1830 — APPROVED** [review]
LGTM

File: testdata/p4tc_samples_outputs/default_action_example_control_blocks.c (hunk 3 above)
Comment (komaljai) [reply testdata/p4tc_samples_outputs/default_action_example_control_blocks.c:223]: This is coming from deparser code in ebpf and cant be removed according to design.

File: testdata/p4tc_samples_outputs/default_action_example_control_blocks.c (hunk 2 above)
Comment (komaljai) [reply testdata/p4tc_samples_outputs/default_action_example_control_blocks.c:150]: This is coming from C code as well and indentation cant be added as we dont know while function processing how much indent to add, This is general function for MethodCallExpression

File: backends/tc/ebpfCodeGen.cpp (hunk 1 above)
Comment (komaljai) [reply backends/tc/ebpfCodeGen.cpp:917]: Addressed

**Pull request merged**